tests/regex/RegexTest.php

```diff
@@ -308,8 +308,10 @@ public function testReplace(
       ->toEqual($expected);
   }
 
-  public static function provideReplaceWith(): varray<mixed> {
-    return varray[
+  /*HHAST_IGNORE_ERROR[DataProviderTypes]
+    `nothing` has fewer tokens than shape(...)*/
```

Comment:
I'm not sure what the problem is here, but isn't this also a limitation of the linter that could be fixed? Ideally, the linter shouldn't force us to change any code that is not actually wrong.

Reply:
My comment is actually wrong, the problem is with the third type, not the second.
`(function(nothing): mixed)` assign to `(function(Regex\Match): string)` fails.
This gets split into the following tokens:
 - `(` assign to `(` :heavy_check_mark: 
 - `function` assign to `function` :heavy_check_mark: 
 - `nothing` assign to `Regex` :heavy_check_mark:
 - `)` assign to `\` :x:

I cannot match two typehints where the token count differs, without building many special cases or building a token parser.
The types are a best effort and we cannot make this work.

Reply:
We can't update the `Regex\Match` to nothing, since that violates `Regex\replace_with()` and we can't change the provider to `Regex\Match`, because then we can't index into the shape blindly.